src/mobile_crawler/domain/providers/lmstudio_adapter.py

```
@@ -0,0 +1,112 @@
+
+        data = {
+            'model': self._model_config.get('model_name', 'local-model'),
+            'messages': messages,
+            'temperature': self._model_config.get('temperature', 0.1)
+        }
+
+        # Local LLMs can be slow, remove timeouts or set very high
+        response = self._session.post(f'{base_url}/chat/completions', json=data, timeout=120)
+        
+        if response.status_code != 200:
+            raise Exception(f"LMStudio API error ({response.status_code}): {response.text}")
```

Ruff (0.15.2)

[warning] Create your own exception

(TRY002)